src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -553,8 +550,7 @@ public void removeDatabaseChangeListener(DatabaseChangeListener l) {
      */
     public static String getResolvedField(String field, BibEntry entry, BibDatabase database) {
         if ("bibtextype".equals(field)) {
-            TypedBibEntry typedEntry = new TypedBibEntry(entry, Optional.ofNullable(database));
-            return typedEntry.getTypeForDisplay();
+            return EntryUtil.capitalizeFirst(entry.getType());
```

Comment:
Yes, but the TypedBibEntry requires the BibDatabaseType. And this is quite cumbersome to pass to this method. So I decided this should suffice for must cases. But it may be problematic still.